R/agglomerate.R

```diff
@@ -154,7 +154,7 @@ setGeneric("mergeFeaturesByRank",
 #' @export
 setMethod("agglomerateByRank", signature = c(x = "SummarizedExperiment"),
     function(x, rank = taxonomyRanks(x)[1], onRankOnly = FALSE, na.rm = FALSE,
-        empty.fields = c(NA, "", " ", "\t", "-", "_"), ...){
+        empty.fields = c(NA, "", " ", "\t", "-", "_"), altexp = NULL, ...){
         # input check
```

Comment:
There should not be reason for adding `altexp` parameter. It should be passed with `...`

Reply:
I will get back to this tomorrow. This is little bit more complicated thing.

Reply:
Because we have to allow MARGIN and altExp. It might be necessary to work with the components of TreeSE instead of the whole TreeSE.

Meaning that:
1. First fetch rowData, tree, assays...
2. agglomerate values
3. Add values back to TreeSE

Or then we can just create another function for MARGIN = 2

Reply:
Currently, we do not have use for agglomerating data column-wise but there might be in the future...

Reply:
ACK

Reply:
I can make a suggestion of this. I have to try different approaches and let's discuss about this after that

Reply:
okay sure.